### DISCUSSION TITLE
How can I back up healthchecks state?

### DISCUSSION THREAD

**Discussion by jdheyburn** [healthchecks/healthchecks#1098] [Q&A] [answered]
Hello!

I'm looking to understand what files and directories I need to backup healthchecks. I see there is a `healthchecks.sqlite` database in the state directory - is it just this that requires backing up?

Thanks!

**Comment by cuu508** [comment, marked as answer]
All* user state (user accounts, projects, checks, received pings, etc.) is stored in the database. If you're using SQLite, then all of it is in the .sqlite file.

[*] Healthchecks supports [offloading large ping bodies to S3-compatible object storage](https://github.com/healthchecks/healthchecks?tab=readme-ov-file#external-object-storage). If you are using that, you may also want to back up the S3 bucket.

**Reply by jdheyburn** [reply]
Thank you! 

Would you mind elaborating on what ping bodies are, and how they would end up being so large?

**Reply by cuu508** [reply]
Sure–when clients make HTTP requests to ping URLs, Healthchecks stores the HTTP request body in the database so users can inspect them later:  https://healthchecks.io/docs/attaching_logs/

By default, Healthchecks stores first 10'000 bytes for each HTTP request, but this can be adjusted: https://healthchecks.io/docs/self_hosted_configuration/#PING_BODY_LIMIT

If your instance processes lots of pings, or the ping bodies are large, it can be operationally easier to store them outside the database – in S3 bucket. This means smaller and faster database backups, less RAM needed to keep all database in memory. But there are downsides too: the user state is now split between database and S3 bucket, and more components means more things that can break.

**Reply by jdheyburn** [reply]
That's cool, I didn't know we could forward logs too. Thanks for the detailed and swift response!
